Packs/ZoomMail/Integrations/ZoomMail/ZoomMail.py

```diff
@@ -417,32 +428,31 @@ def fetch_incidents(client: ZoomMailClient, params: dict[str, str]) -> None:
 
         # Check if it's a new message, and if it's either all messages (fetch_threads is False) or only thread starters
         if (internal_date > last_fetch_info["internalDate"] or (
-            internal_date == last_fetch_info["internalDate"] and message_id not in last_fetched_ids
+            internal_date == last_fetch_info["internalDate"] and message_id not in last_fetch_info["ids"]
         )) and (fetch_threads or (not fetch_threads and message_id == thread_id)):
             incident = zoom_mail_to_incident(message_details, client, fetch_from)
             incidents.append(incident)
-            message_dates.append(internal_date)
+            if not last_page_fetch_token:
+                message_dates.append(internal_date)
 
-    if message_dates:
+    if message_dates and not next_page_token:
         new_internal_date = max(message_dates)
-        newly_fetched_ids = [
-            msg["id"] for msg in messages
+        new_ids = [
+            msg["id"] for msg in messages if msg["internalDate"] == new_internal_date
         ]
 
-        last_fetched_ids.extend(newly_fetched_ids)
-
-        last_fetch_info = {
-            "internalDate": new_internal_date,
-            "ids": last_fetched_ids
-        }
+        last_fetch_info = {"internalDate": new_internal_date, "ids": new_ids}
 
     # If we don't collect any new messages, but we have a next fetch token,
     # we are fetching old messages and need to stop.
-
     if len(incidents) == 0 and next_page_token:
         demisto.info("No new messages found, stopping fetch.")
         next_page_token = None
 
+    # If we dont have a token, we are fetching new messages and need to update the last fetch info
+    if next_page_token is None:
+        last_fetch_info["internalDate"] = now_epoch
```

Comment:
I think that internalDate should never be now_epoch.
there are 3 cases for internalDate.
1. if it is the first run, use first run calculation
2. if there are new mail, take the last mail time make sure it is later then current internalDate
3. if there are no mails, preserve internalDate.